stats/stats.go

```diff
@@ -393,21 +390,56 @@ func (h *Histogram) Clone() *Histogram {
 // CopyFrom sets the content of this object to a copy of the src.
 func (h *Histogram) CopyFrom(src *Histogram) {
 	h.Counter = src.Counter
-	// we don't copy offset/divider as this assumes compatible src/dest
-	for i := 0; i < len(h.Hdata); i++ {
-		h.Hdata[i] += src.Hdata[i]
+	h.CopyHDataFrom(src)
+}
+
+func (h *Histogram) CopyHDataFrom(src * Histogram)  {
+	if h.Divider == src.Divider && h.Offset == src.Offset {
+		for i := 0; i < len(h.Hdata); i++ {
+			h.Hdata[i] += src.Hdata[i]
+		}
+		return;
 	}
+	//Scale values are not equal for histograms.
+	//Need to receive record values from src and add them to h according to h's offset and divider
+	var scaledValue float64
+	for i := 0; i < len(src.Hdata); i++ {
+		if src.Hdata[i] == 0 {
+			continue
+		}
+		scaledValue = -1
+		//Find scaleValue in val2Bucket
+		for j := 0; j<numBuckets; j++ {
+			if val2Bucket[j] == i {
+				scaledValue = float64(j)
+			}
+		}
+		if scaledValue == -1{
+			continue
+		}
+		v := scaledValue * src.Divider + src.Offset
+		for src.Hdata[i] != 0{
+			h.appendRecordToHistogram(v)
+			src.Hdata[i] -= 1
+		}
+	}
+}
+
+func (h*Histogram) appendRecordToHistogram(v float64)  {
```

Comment:
can you move that function to line 215, it would minimize the diff

Reply:
Sure